tests/cc_testing_utils.py

```diff
@@ -20,105 +16,39 @@ def generate_random_graph(graph_size, seed=None):
     return graph
 
 
-def register_cc_df(G):
-    from tests.basic_settings import get_settings_dict
+def nodes_and_edges_from_graph(G):
+    edges = nx.to_pandas_edgelist(G)
+    edges.columns = ["unique_id_l", "unique_id_r"]
 
-    settings_dict = get_settings_dict()
+    nodes = pd.DataFrame({"unique_id": G.nodes})
 
-    df = nx.to_pandas_edgelist(G)
-    df.columns = ["unique_id_l", "unique_id_r"]
-    df_concat = pd.concat(
-        [pd.DataFrame({"unique_id_l": G.nodes, "unique_id_r": G.nodes}), df]
-    )
+    return nodes, edges
 
-    # boot up our linker
-    table_name = "__splink__df_predict_graph"
-    # this registers our table under __splink__df__{table_name}
-    # but our cc function actively looks for "__splink__df_predict"
-    db_api = DuckDBAPI()
-
-    linker = Linker(
-        df_concat, settings_dict, input_table_aliases=table_name, db_api=db_api
-    )
-
-    # re-register under our required name to run the CC function
-    linker.table_management.register_table(df_concat, table_name, overwrite=True)
-
-    df_nodes = pd.DataFrame({"unique_id": G.nodes})
-    linker.table_management.register_table_input_nodes_concat_with_tf(df_nodes)
-
-    # add our prediction df to our list of created tables
-    predict_df = DuckDBDataFrame(table_name, table_name, db_api)
-
-    return linker, predict_df
-
-
-def run_cc_implementation(linker, predict_df):
-    pipeline = CTEPipeline()
-    concat_with_tf = compute_df_concat_with_tf(linker, pipeline)
 
+def run_cc_implementation(nodes, edges):
     # finally, run our connected components algorithm
-    cc = solve_connected_components(
-        linker,
-        predict_df,
-        concat_with_tf=concat_with_tf,
-        _generated_graph=True,
+    db_api = DuckDBAPI()
+    cc = cluster_pairwise_predictions_at_threshold(
+        nodes=nodes,
+        edges=edges,
+        db_api=db_api,
+        node_id_column_name="unique_id",
+        edge_id_column_name_left="unique_id_l",
+        edge_id_column_name_right="unique_id_r",
+        threshold_match_probability=None,
     ).as_pandas_dataframe()
+
     cc = cc.rename(columns={"unique_id": "node_id", "cluster_id": "representative"})
     cc = cc[["node_id", "representative"]]
+    cc.sort_values(by=["node_id", "representative"], inplace=True)
     return cc
 
 
-def benchmark_cc_implementation(linker_df):
-    # add a schema so we don't need to re-register our df
-    linker_df.db_api._con.execute(
-        """
-        create schema if not exists con_comp;
-        set schema 'con_comp';
-        """
-    )
-
-    df = run_cc_implementation(linker_df)
-    linker_df.db_api._con.execute("drop schema con_comp cascade")
-
-    return df
-
-
 def networkx_solve(G):
     rows = []
     for cc in cc_nx(G):
         m = min(list(cc))
         for n in cc:
             row = {"node_id": n, "representative": m}
             rows.append(row)
-    return pd.DataFrame(rows)
-
-
-def check_df_equality(df1, df2, skip_dtypes=False):
```

Comment:
syntax like `assert (cc_df.values == nx_df.values).all()` is sufficient so this doesn't need to be a fn